internal/resources/ekscluster/ekscluster_flatten_test.go

```diff
@@ -472,20 +649,24 @@ func TestFlattenConfig(t *testing.T) {
 	}
 }
 
-func getClusterSpec() (*eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec, []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolDefinition) {
-	return &eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec{
-			ClusterGroupName: "test-cg",
-			ProxyName:        "test-prooxy",
-			Config:           getConfig(),
-		}, []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolDefinition{
-			{
-				Info: &eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolInfo{
-					Description: "test np",
-					Name:        "test-np",
-				},
-				Spec: getNodepoolSpec(),
+func getClusterSpec(addonsConfig *eksmodel.VmwareTanzuManageV1alpha1EksclusterAddonsConfig) (*eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec, []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolDefinition) {
```

Comment:
Let this method return a full cluster spec. For the cases where we don't need the addon config, we can overwrite it in the `getInput` function as done for the "empty config" test case.

Reply:
@asutoshpalai I have made changes as requested